cmd/heartbeats/main.go

```diff
@@ -115,10 +115,13 @@ func main() {
 	}
 
 	var period time.Duration
-	if p, err := strconv.Atoi(periodStr); err != nil {
-		period = time.Duration(5) * time.Second
-	} else {
+	// try to parse as a duration, then as an int, then default to 5s
+	if p, err := time.ParseDuration(periodStr); err == nil {
+		period = p
+	} else if p, err := strconv.Atoi(periodStr); err == nil {
 		period = time.Duration(p) * time.Second
+	} else {
+		period = time.Duration(5) * time.Second
```

Comment:
I would rather panic to catch the error early on.

Reply:
Do you mean something like 

```go
if periodStr == "" {
  period = 5 * time.Second
} else {
  log.Fatalf("Invalid period interval provided: %q", periodStr)
}
```
?

Reply:
yup